BackEndApp/authv1/auth.py

```diff
@@ -53,3 +57,50 @@ def verify(self):
             return True
 
         return False
+
+
+class Otp:
+    def __init__(self, user, otp=None, expire=None, **kwargs):
+        self.user = user
+        self.username = user.get("username")
+        self.otp = otp
+        self.expire = expire
+        self.db = connect()
+        self.collection_name = kwargs.get("collection", "otp")
+        self.collection = self.db[self.collection_name]
+        self.attributes = kwargs
+
+    def create(self, time_delta=5):
+        try:
+            """
+            time_delta: (int) In minutes
+            """
+            print("creating otp")
+            self.expire = datetime.now() + timedelta(minutes=time_delta)
+            self.otp = "".join(
+                random.SystemRandom().choice(string.ascii_uppercase + string.digits)
+                for _ in range(6)
+            )
+            doc_otp = {
+                "user": self.user,
+                "otp": self.otp,
+                "expire": self.expire,
+                "username": self.user["username"],
+            }
+            self.collection.insert_one(doc_otp)
+            return self.otp
+        except Exception as e:
+            return None
+
+    def find(self):
+        return self.collection.find_one({"username": self.username})
+
+    def verify(self):
```

Comment:
You need to check the OTP from MongoDB having this username, and whether it is same or not.